src/box/txn.h

```diff
@@ -777,6 +780,52 @@ txn_begin_stmt(struct txn *txn, struct space *space, uint16_t type);
 int
 txn_begin_in_engine(struct engine *engine, struct txn *txn);
 
+static inline bool
+txn_is_linearizable(const struct txn *txn)
+{
+	return txn->isolation == TXN_ISOLATION_LINEARIZABLE;
+}
+
+/**
+ * Check if a space supports linearizable access, i.e. it is either synchronous
+ * or not replicated at all, and its engine prevents dirty reads.
+ */
+static inline int
+txn_check_space_linearizability(const struct txn *txn,
+				const struct space *space)
+{
+	assert(txn_is_linearizable(txn));
+	(void)txn;
+	/*
+	 * XXX: can't promise the user linearizability with Vinyl until
+	 * https://github.com/tarantool/tarantool/issues/5522 is fixed.
```

Comment:
I'm planning to fix #5522 soon. Let's file another ticket for the reference and add a TODO instead of XXX.

```C
/* TODO(gh-XXXX): Enable linearizable transactions in Vinyl. */
```

Reply:
Here: https://github.com/tarantool/tarantool/issues/7699.
Changed comment is in one of the diffs below.